File: eventbus/eclipse_java/eventbus/src/orbisoftware/msgblit/eventbus/unittest/EventEmitterExample.java
package orbisoftware.msgblit.eventbus.unittest;

import orbisoftware.msgblit.eventbus.EBEvent;
import orbisoftware.msgblit.eventbus.EBRouter;

public class EventEmitterExample extends Thread {

   EventEmitterExample(EBRouter router, int emitterID) {

      this.router = router;
      this.emitterID = emitterID;
   }

   public void run() {

      while (true) {

         EBEvent event = new EBEvent();

         event.eventID = emitterID << 2;
         event.eventCatType = 20;
         event.eventDefType = 25;
         event.eventData = "java event data";

         if (emitterID == 1)
            router.publishEvent(event, EBRouter.DistType.SHARED);
         else
            router.publishEvent(event, EBRouter.DistType.LOCAL);

         try {
            Thread.sleep(1000);
         } catch (InterruptedException e) {
         }
      }
   }

   private EBRouter router;
   private int emitterID;
}

File: eventbus/eclipse_cpp/eventbus/src/unittest/EventEmitterExample.cpp
#include <EventEmitterExample.hpp>
#include <boost/thread.hpp>
#include <string>

EventEmitterExample::EventEmitterExample(EBRouter* router, int emitterID) {

   this->router = router;
   this->emitterID = emitterID;
}

void EventEmitterExample::start() {
   boost::thread(&EventEmitterExample::run, this);
}

void EventEmitterExample::run() {

   while (true) {

      EBEvent event;

      event.eventID = emitterID << 4;
      event.eventCatType = 11;
      event.eventDefType = 12;
      event.eventData = "C++ event data";

      if (emitterID == 1)
         router->publishEvent(event, EBRouter::SHARED);
      else
         router->publishEvent(event, EBRouter::LOCAL);

      boost::this_thread::sleep(boost::posix_time::milliseconds(1000));
   }
}

